### DISCUSSION TITLE
disable escaping

### DISCUSSION THREAD

**Discussion by aandis** [syntax-tree/unist#87] [Q&A]
Given following simple code to parse markdown with parser and outputting it back as markdown

```js
const unified = require('unified');
const parser = require('remark-parse');
const toMarkdown = require('mdast-util-to-markdown');

let content = `Hello [[World]]`
const ast = await unified().use(parser).parse(content);
const markdown = toMarkdown(ast);
console.log(markdown);
```

the output is 
```md
Hello \[[World]]
```

But the output should be `Hello [[World]]`. How do I get rid of the leading `\`?

**Comment by ChristianMurphy** [comment]
Welcome @aandis!

Taking a step back, `remark` is an AST ecosystem.
Meaning it will preserve the structure/presentation of the content, but will not retain the exact characters.
You can read more about this is some previous discussions on AST vs CST.
https://github.com/remarkjs/remark/search?q=cst&type=issues and https://github.com/orgs/remarkjs/discussions?discussions_q=cst

Coming back to your questions, the solution comes down to what are you trying to do?
Reading between the lines of your comment you could be trying to do one of two things.
1. Display `[[World]]` as plain text in the final result. And in this case the escaped output is fine, it doesn't matter if it is escaped, the final presentation will be the same.
2. Display a wikilink or shortcode, these are not standard markdown, and need a plugin to be parsed and handled correctly. Take a look at the list of plugins for options: https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins
   * You may be interested in https://github.com/landakram/remark-wiki-link
   * Or in https://github.com/djm/remark-shortcodes

**Reply by aandis** [reply]
> what are you trying to do?

@ChristianMurphy I'm trying to parse a markdown file as an ast, tweak the AST to modify the markdown content and then render it back into markdown to feed it into [Docusaurus](https://docusaurus.io/) which finally converts it into HTML.

The problem is that when the markdown output from unified is `Hello \[[World]]`, the end html result is also `Hello \[[World]]` which is not what we want because the source was `Hello [[World]]`.

> And in this case the escaped output is fine, it doesn't matter if it is escaped, the final presentation will be the same.

I'm curious why though? It depends on how the output is being rendered I think? `Hello \[[World]]` is different from `Hello [[World]]` both as text and HTML.

**Reply by wooorm** [reply]
Please see 2a for the proper solution, in Christian’s post.

The thing is: wikilinks are not markdown. GFM is not markdown. Frontmatter is not markdown.
If you want those things, you need to enable them. You can enable them with a plugin.

**Reply by aandis** [reply]
hi @wooorm. I'm not trying to render wikilinks. I just want to render `Hello [[World]]` _as is_ in the markdown file without anything being escaped, so that when it's converted to html, it still displays as `Hello [[World]]`. Hope that makes sense.

**Reply by aandis** [reply]
`[[World]]` has no special meaning in my content. It's just plain text content and I want to be able to render it as is after passing it through a unified pipeline.

**Reply by wooorm** [reply]
That looks a lot like a wikilink? What do you think it is?
If it’s not a wikilink, then the escape renders fine and you don’t have any problems:

```markdown
Hello [[World]]
Hello \[[World]]
```

Yields:

Hello [[World]]
Hello \[[World]]


I am not talking about rendering wikilinks (as in, going to HTML). I believe you need to add support for *parsing* these things.

**Reply by aandis** [reply]
> then the escape renders fine

apologies if I'm not explaining well since the escape doesn't render fine. When you convert a markdown file containing `Hello \[[World]]` to html as done in Docusaurus, it renders as `Hello \[[World]]` and not `Hello [[World]]`. This is what I want to fix.

**Reply by wooorm** [reply]
Then that is a bug in docusaurus.

**Reply by ChristianMurphy** [reply]
Adding on to

> Then that is a bug in docusaurus.

when the original example is rendered with plain remark: https://codesandbox.io/s/agitated-kepler-crsc2p
or plain mdx: https://codesandbox.io/s/lively-pine-hyutg1?file=/src/App.js

Note that in both cases, the source has one example escaped and one not escaped, but the results of both are identical when rendered.
Meaning if you see an extra `/` rendered it is something outside of `unist`, `mdast`, `unified`, `remark`, `rehype`, or `mdx` causing the issue.
Most likely in customization added by docusaurus itself.

**Reply by ninja7dev** [reply]
@wooorm or others: Can we make an option to disable the escaping, if needed? I'm here for a similar issue.

**Reply by wooorm** [reply]
please stop spamming and replying everywhere. Stick to one place. Open one single issue. I’ll block you otherwise.
